packages/indexer-cli/src/command-helpers.ts

```diff
@@ -206,3 +207,47 @@ export function suggestCommands(
   )
   return suggestions.length > 0 ? suggestions : supported_commands
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function extractProtocolNetworkOption(
+  options: {
+    [key: string]: any
+  },
+  required = false,
+): string | undefined {
+  const { n, network } = options
+
+  // Tries to extract the --network option from Gluegun options.
+  // Throws if required is set to true and the option is not found.
+  if (!n && !network) {
+    if (required) {
+      throw new Error("The option '--network' is required")
+    } else {
+      return undefined
+    }
+  }
+
+  // Check for invalid usage
+  const allowedUsages =
+    (n === undefined && typeof network === 'string') ||
+    (network === undefined && typeof n === 'string')
+  if (!allowedUsages) {
+    throw new Error("Invalid usage of the option '--network'")
+  }
+  const input = (network ?? n) as string
+
+  try {
+    return validateNetworkIdentifier(input)
```

Comment:
Would it make sense to merge `extractProtocolNetworkOption()` together with `validateNetworkIdentifier()`?